lib/ansible/executor/playbook_executor.py

```diff
@@ -75,7 +75,7 @@ def run(self):
         entry = {}
         try:
             for playbook_path in self._playbooks:
-                pb = Playbook.load(playbook_path, variable_manager=self._variable_manager, loader=self._loader)
+                pb = Playbook.load(playbook_path, variable_manager=self._variable_manager, loader=self._loader, executor=self)
```

Comment:
no need to pass playbook_executor, callback already has the cli options/config and can check if it was run in check_mode `self._options['check_mode']`

Reply:
Unfortunately this is not the case (as checked with pdb).
Possibly the cause is the way callbacks are loaded in 

lib/ansible/executor/task_queue_manager.py

```
                    self._stdout_callback.set_options()
```

The other path from callback base seems to not work, as cli is None

lib/ansible/plugins/callback/__init__.py

```
        if cli:
            self._options = cli.options
        else:
```

Reply:
'cli' should be loaded, otherwise that is a bug. the `set_options` there refers to plugin_options not 'options' (sorry for name overload, but one is general across plugins and i forgot callbacks already were using it for cli options.

in any case i'm in the process of removing play_context from option handling so it is not a good idea to rely on it

Reply:
Regarding your second though (that play_context would be decoupled from options) 
- We could either keep somehow the link between playbook Executor and callback (in order to access run time option) and disregard the _play_context. Indeed I could have used playbook executor for both the CHECK MODE message code above and the DRY RUN one.

- Or (taking into account the second your first observation) the None cli bug should be investigated. I think I should leave this to some more experienced with ansible internals.

Reply:
Now the refs to playbookExecutor and play_context have been removed, and we use the newly introduced context.